diff -druN Safe.sol Safe.sol
--- Safe.sol	2023-08-16 17:09:10
+++ Safe.sol	2023-08-25 10:46:24
@@ -75,7 +75,7 @@
          * so we create a Safe with 0 owners and threshold 1.
          * This is an unusable Safe, perfect for the singleton
          */
-        threshold = 1;
+        // threshold = 1; MUNGED: remove and add to constructor of the harness
     }
 
     /**
diff -druN base/Executor.sol base/Executor.sol
--- base/Executor.sol	2023-08-16 17:09:10
+++ base/Executor.sol	2023-08-25 10:46:24
@@ -26,11 +26,8 @@
         uint256 txGas
     ) internal returns (bool success) {
         if (operation == Enum.Operation.DelegateCall) {
-            // solhint-disable-next-line no-inline-assembly
-            /// @solidity memory-safe-assembly
-            assembly {
-                success := delegatecall(txGas, to, add(data, 0x20), mload(data), 0, 0)
-            }
+            // MUNGED lets just be a bit more optimistic, `execute` does nothing for `DELEGATECALL` and always returns true
+            return true;
         } else {
             // solhint-disable-next-line no-inline-assembly
             /// @solidity memory-safe-assembly
